LGTM deprecation: Update basic queries to use VS Code

felicitymay opened the pull request:
This pull request updates the basic query introduction article for Python to use VS Code. I've raised this for discussion. Once we've agreed the updates, I'll apply them to the other basic query introductions.

felicitymay commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -3,7 +3,9 @@
-Learn to write and run a simple CodeQL query using LGTM.
+Learn to write and run a simple CodeQL query using Visual Studio Code with the CodeQL extension.
+
+.. include:: ../reusables/setup-to-run-queries.rst
```
I've added this placeholder here so that it's easy to update this as we steadily make it easier to get started with running CodeQL queries.

felicitymay commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -14,62 +16,65 @@ The query we're going to run performs a basic search of the code for ``if`` stat
+#. Click **From GitHub** or the GitHub logo |github-db| at the top of the CodeQL extension to open an entry field.
 
-   .. pull-quote::
+#. Copy the URL for the repository into the field and press the keyboard **Enter** key.
```
❓ Do we want to suggest an example repository here? For example, https://github.com/protocolbuffers/protobuf.

jf205 replied:
Yes, I think that's quite helpful 😄

jf205 commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -14,62 +16,65 @@ The query we're going to run performs a basic search of the code for ``if`` stat
-#. Click **Run**.
+#. Create a ``qlpack.yml`` file in your experiments folder with the contents shown below. This tells the CodeQL extension that any queries you create in the folder are intended to run on C or C++ CodeQL databases.
 
-   The name of the project you are querying, and the ID of the most recently analyzed commit to the project, are listed below the query box. To the right of this is an icon that indicates the progress of the query operation:
+   .. code-block:: yaml
 
-   .. image:: ../images/query-progress.png
-       :align: center
+      name: github-owner/cpp-experiments
+      version: 0.0.1
+      dependencies:
+         codeql/cpp-all: ^0.1.2
 
-   .. pull-quote::
+#. Create a second new file in your experiements folder with the ``.ql`` file extension. You will write your first query in this file.
 
```
For simplicity, we could point users to the quick query command here: https://codeql.github.com/docs/codeql-for-visual-studio-code/analyzing-your-projects/#running-a-quick-query

It would save folks having to create a new qlpack.yml file. However, they would have to copy the contents of the synthetic CodeQL pack to somewhere permanent if they wanted to retain their test queries, but that could be an optional extra step at the end of this procedure (and we could simply link to the docs).

felicitymay commented:
@jf205 - thanks for your review comments ✨ 

I've updated the article to use quick query and added a note that they should move the whole directory if they want to store the query somewhere more permanent. Could you take a second look? I'd like to be sure that we're both happy before applying this approach to the other similar articles.

jf205 commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -134,14 +141,38 @@ To exclude ``if`` statements that have an ``else`` branch:
+      name: github-owner/cpp-experiments
+      version: 0.0.1
+      dependencies:
+         codeql/cpp-all: ^0.1.2
```
I'm not up to date withe core library pack versioning, but do you know if would `codeql/cpp-all: *` (latest version) suffice?

jf205 commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -134,14 +141,38 @@ To exclude ``if`` statements that have an ``else`` branch:
+      dependencies:
+         codeql/cpp-all: ^0.1.2
+
+#. Create a second new file in your experiements folder with the ``.ql`` file extension. You will write your first query in this file.
```
```suggestion
#. Create a second new file in your experiments folder with the ``.ql`` file extension. You can write your next query in this file. You can also copy any of your quick queries from the current session over to the ``cpp-experiments`` folder for future development. 
```

jf205 commented on docs/codeql/codeql-language-guides/basic-query-for-cpp-code.rst:
```diff
@@ -134,14 +141,38 @@ To exclude ``if`` statements that have an ``else`` branch:
    There are now fewer results because ``if`` statements with an ``else`` branch are no longer reported.
 
-➤ `See this in the query console <https://lgtm.com/query/1899933116489579248/>`__
+Saving your quick query for future development
```
If we can get away with specifying the latest version of the library packs, then maybe this could be turned into a reusable snippet (with a placeholder <lang> instead of cpp as appropriate)?

felicitymay replied:
Many thanks for the review.

🤦🏻‍♀️ I copied this here when I started updating the procedure, and then realized that people could just move/copy the `Quick Query` folder created by the extension to a permanent location, as described on line 79.

I meant to delete this section from line 148 to 164 before committing the changes. Unless you think this would be useful, I suggest that we delete this.

jf205 replied:
🔥  (please delete away!)

jf205 reviewed (commented):
Thanks for the updates @felicitymay ✨ . I have a few questions but otherwise this is looking in good shape.

felicitymay commented:
My apologies for all the pings. I clearly managed to mess something up when I updated my local branch. Closing.
